src/main.tsx

```diff
@@ -2,9 +2,12 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 
 import { App } from './App';
+import { UserProvider } from './hooks/useLoggedInUser';
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
 	<React.StrictMode>
-		<App />
+		<UserProvider>
```

Comment:
Toto jsem udělal tak napůl. Líbilo by se mi, kdyby byly všichni globální Provideři v jedné komponentě (asi zde v `main.tsx`), tzn. že by se sem ještě musel přesunout `ThemeProvider` společně s `CssBaseline`. V `App.tsx` by pak zůstala samotná appka (`BrowserRouter`). Co si o tom myslíš, @Antalyan?

Reply:
Ok `<UserProvider>` přesunu do `App.tsx`, ať je to na jednom místě. A pak uvidíme, jak se nám to nafoukne a podle toho rozdělíme.